src/main/java/org/folio/search/service/browse/CallNumberBrowseService.java

```diff
@@ -84,6 +80,29 @@ protected String getValueForBrowsing(CallNumberBrowseItem browseItem) {
     return browseItem.getShelfKey();
   }
 
+  private BrowseResult<CallNumberBrowseItem> getPrecedingResult(BrowseRequest request,
+                                                                BrowseContext context,
+                                                                SearchSourceBuilder precedingQuery,
+                                                                MultiSearchResponse multiSearchResponse) {
+    var responses = multiSearchResponse.getResponses();
+    var precedingResult = callNumberBrowseResultConverter.convert(responses[0].getResponse(), context, false);
+    var succeedingPrecedingResult = callNumberBrowseResultConverter.convert(responses[1].getResponse(), context, false);
+
+    if (precedingResult.getRecords().isEmpty() && precedingResult.getTotalRecords() > 0) {
+      log.debug("getPrecedingResult:: preceding result are empty: Do additional requests");
+      precedingResult = additionalPrecedingRequests(request, context, precedingQuery);
+    }
+
+    if (!succeedingPrecedingResult.isEmpty()) {
+      log.debug("getPrecedingResult:: succeeding preceding result are not empty: Update preceding result");
```

Comment:
I just can't understand what is "succeeding preceding"
```suggestion
      log.debug("getPrecedingResult:: backward preceding result are not empty: Update preceding result");
```